### PR TITLE
feat(python): throw HTTPExceptions instead of IOException for http errors

### PR DESCRIPTION
**Pull request by Mause**
I've pulled out the NodeJS equivalent, will submit it once https://github.com/duckdb/duckdb/pull/6434 is merged

Feedback on C++ code very welcome!

### CODE REVIEW

File: tools/pythonpkg/tests/conftest.py [hunk 1]
```diff
@@ -1,33 +1,34 @@
...
     return cursor
- 
+
 
 @pytest.fixture(scope="function")
 def require():
     def _require(extension_name, db_name=''):
         # Paths to search for extensions
+
+        build = normpath(join(dirname(__file__), "../../../build/"))
```
Comment (Tishj) [inline tools/pythonpkg/tests/conftest.py:21]: Be careful about changing this, I have done this in the past and found out the hard way that the paths aren't the same in the CI.
I also added something similar in a local change yesterday, but replacing all the others seems scary

File: tools/pythonpkg/tests/extensions/test_httpfs.py [hunk 2]
```diff
@@ -1,12 +1,17 @@
 import duckdb
 import os
 import pandas as pd
+from pytest import raises, mark
+
+
+# We only run this test if this env var is set
+pytestmark = mark.skipif(
+    not os.getenv('DUCKDB_PYTHON_TEST_EXTENSION_REQUIRED', False),
+    reason='httpfs extension not available'
```
Comment (Tishj) [inline tools/pythonpkg/tests/extensions/test_httpfs.py:10]: Isn't this already done by `require` ?
I recently learned you can actually see these reasons when you run pytest with the `-rs` flags. (hardly documented anywhere for some reason)

**Review by Tishj — COMMENTED** [review]
Looks like a great change, saving the response + error code is definitely a good idea!

File: tools/pythonpkg/tests/extensions/test_httpfs.py (hunk 2 above)
Comment (Mause) [reply tools/pythonpkg/tests/extensions/test_httpfs.py:10]: Yeah I'm not sure why this environment variable exists, happy to remove this check though

File: tools/pythonpkg/tests/conftest.py (hunk 1 above)
Comment (Tishj) [reply tools/pythonpkg/tests/conftest.py:21]: Then it's probably fine, happy to remove this gunk if nothing breaks :thumbsup:

**Comment by Mytherin** [comment]
Thanks! LGTM

**Pull request merged**